### PR TITLE
add forward-exec plugin

### PR DESCRIPTION
**Pull request by tgunsch**
Hi,

not sure if this really worth a plugin and useful for others.

For our integration and functional tests we often needed a smoke test, where we need to

* create a port-forward to a pod /service
* execute a curl command to test the pod or get data from them.
  * for security reasons we only have scratch images without a curl command, so we execute the curl locally / in CI system
* finally we drop the port-forward.

Until now we had a simple helper script to do this. Now I have experimented with kubectl plugins, so I converted the script to a go plugin.

Example usage:
```
kubectl forward-exec svc/my-service 8080:80 -- curl http://localhost:8080/validate-pod
```

### CODE REVIEW

**Comment by ahmetb** [comment]
:robot: _Beep beep! I’m a robot speaking on behalf of @ahmetb._ :robot:

-----

Thanks for submitting your kubectl plugin to Krew!
One of the krew-index maintainers will review it soon. Note that the reviews for new plugin submissions may take a few days.

In the meanwhile, here are a few tips to make your plugin manifest better:
* Make sure your plugin follows the [best practices](https://krew.sigs.k8s.io/docs/developer-guide/develop/best-practices/).
* Eliminate redundant wording form `shortDescription` (it should be max 50 characters).
* Try to word wrap your `description` to 80-character lines (no usage examples, please).

Thanks for your patience!
/kind new-plugin

**Review by ahmetb — COMMENTED** [review]
I'm not entirely sure about the usefulness of this plugin, can you please check we already don't have a plugin along these lines?

Technically this can be a one-liner like:

  port-forward &; pid=$! while:; do nc [...]; done; CMD; kill $pid

and we prefer not to have a lot of these one-liner utilities as they go stale and users should err on the side of composability of core utilities on their shell.

/kind gray-area

**Comment by tgunsch** [comment]
> I'm not entirely sure about the usefulness of this plugin, can you please check we already don't have a plugin along these lines?
> 
> Technically this can be a one-liner like:
> 
> port-forward &; pid=$! while:; do nc [...]; done; CMD; kill $pid
> 
> and we prefer not to have a lot of these one-liner utilities as they go stale and users should err on the side of composability of core utilities on their shell.
> 
> /kind gray-area

Hi @ahmetb,

thanks for your input. We searched already but didn't find a plugin with comparable feature.

You are right, this could be achieved with an one-liner. We still think it's better to handle the
"wait until tunnel is established" in the port-forward "part", than doing an active wait in the executed command.
(IMHO it's also faster to type :-)

We just use this in our own CI pipeline and just thought this would also be helpful to others. If you don't think so, feel free to close this PR.

Kind regards
Thomas

File: plugins/forward-exec.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: forward-exec
+spec:
+  version: "v0.2.0"
+  homepage: https://github.com/tgunsch/kubectl-forward-exec
+  shortDescription: "Forward a port to a pod, run a command on local machine and finally stop the port forward."
```
[the remark this answers is not in the document]
Comment (tgunsch) [reply plugins/forward-exec.yaml]: Fixed

File: plugins/forward-exec.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  description: |
+    This plugin forwards a port to a pod, run a command (e.g. curl) on local machine and finally stop the port forward.
```
[the remark this answers is not in the document]
Comment (tgunsch) [reply plugins/forward-exec.yaml:10]: Fixed

**Pull request closed without merging**